File: patterns/java/src/refactoring_guru/patterns/chain_of_responsibility/user_authenticator/middleware/RoleCheckMiddleware.java
package refactoring_guru.patterns.chain_of_responsibility.user_authenticator.middleware;

/**
 * RU: Конкретный элемент цепи обрабатывает запрос по-своему.
 */
public class RoleCheckMiddleware extends Middleware {
    public boolean check(String email, String password) {
        if (email.equals("admin@example.com")) {
            System.out.println("Hello, admin!");
            return true;
        }
        System.out.println("Hello, user!");
        return checkNext(email, password);
    }
}

File: patterns/cs/src/RefactoringGuru/ChainOfResponsibility/Example/Middleware/RoleCheckMiddleware.cs
using System;

namespace RefactoringGuru.ChainOfResponsibility.Example.Middleware
{
    /**
     * RU: Конкретный элемент цепи обрабатывает запрос по-своему.
     */
    class RoleCheckMiddleware : Middleware
    {
        public override bool Check(string email, string password)
        {
            if (email.Equals("admin@example.com"))
            {
                Console.WriteLine("Hello, admin!");
                return true;
            }
            Console.WriteLine("Hello, user!");
            return CheckNext(email, password);
        }
    }
}
